dnf5-plugins/config-manager_plugin/addrepo.cpp

```diff
@@ -0,0 +1,462 @@
+    save_filename_opt->set_long_name("save-filename");
+    save_filename_opt->set_description(
+        "Set the name of the configuration file of the added repository. The \".repo\" extension is added if it is "
+        "missing.");
+    save_filename_opt->set_has_value(true);
+    save_filename_opt->set_arg_value_help("FILENAME");
+    save_filename_opt->set_parse_hook_func([this](ArgumentParser::NamedArg *, const char *, const char * value) {
+        save_filename = value;
+        return true;
+    });
+    cmd.register_named_arg(save_filename_opt);
+
+    auto set_opt = parser.add_new_named_arg("set");
+    set_opt->set_long_name("set");
+    set_opt->set_description("Set option in newly created repository");
+    set_opt->set_has_value(true);
+    set_opt->set_arg_value_help("REPO_OPTION=VALUE");
+    set_opt->set_parse_hook_func(
+        [this](
+            [[maybe_unused]] ArgumentParser::NamedArg * arg, [[maybe_unused]] const char * option, const char * value) {
+            auto val = strchr(value + 1, '=');
+            if (!val) {
+                throw libdnf5::cli::ArgumentParserError(
+                    M_("set: Badly formatted argument value \"{}\""), std::string{value});
+            }
+            std::string key{value, val};
+            std::string key_value{val + 1};
+
+            // Test if the repository option can be set.
+            try {
+                tmp_repo_conf.opt_binds().at(key).new_string(libdnf5::Option::Priority::RUNTIME, key_value);
+            } catch (const libdnf5::Error & ex) {
+                throw ConfigManagerError(
+                    M_("Cannot set repository option \"{}={}\": {}"), key, key_value, std::string{ex.what()});
+            }
+
+            // Save the repo option for later writing to a file.
+            const auto [it, inserted] = repo_opts.insert({key, key_value});
+            if (!inserted) {
+                if (it->second != key_value) {
+                    throw ConfigManagerError(
+                        M_("Sets the \"{}\" option again with a different value: \"{}\" != \"{}\""),
+                        key,
+                        it->second,
+                        key_value);
+                }
+            }
+            return true;
+        });
+    cmd.register_named_arg(set_opt);
+
+    // Set conflicting arguments
+    repo_id_opt->add_conflict_argument(*from_repofile_opt);
+    set_opt->add_conflict_argument(*from_repofile_opt);
+}
+
+
+void ConfigManagerAddRepoCommand::configure() {
+    auto & ctx = get_context();
+    auto & base = ctx.base;
+    auto logger = base.get_logger();
+
+    const auto & repo_dirs = base.get_config().get_reposdir_option().get_value();
+    if (repo_dirs.empty()) {
+        throw ConfigManagerError(M_("Missing path to repository configuration directory"));
+    }
+
+    std::filesystem::path repo_dir = repo_dirs.front();
+
+    std::string url;
+    if (!repofile_url.empty()) {
+        url = repofile_url;
```

Comment:
A suggestion - What about to make the code more condense?
```
    if (!repofile_url.empty()) {
        add_repo_from_repofile(repofile_url, repo_dir);
        return;
```

Reply:
Reworked.